packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/simulationservice/interventions/Intervention.java

```diff
@@ -32,4 +35,43 @@ public Intervention clone() {
 		}
 		return intervention;
 	}
+
+	// Check for no negative timesteps
+	// Check for no duplicate time + appliedTo pairs in static interventions
+	public Boolean validateIntervention() throws Exception {
+		//Sort static list
+		Collections.sort(this.staticInterventions, (staticOne, staticTwo) -> {
+			final String stringOne = staticOne.getAppliedTo() + staticOne.getTimestep().toString();
+			final String stringTwo = staticTwo.getAppliedTo() + staticTwo.getTimestep().toString();
+			return stringOne.compareTo(stringTwo);
+		});
+
+		for (int i = 0; i < this.staticInterventions.size(); i++) {
+			final Number time = this.staticInterventions.get(i).getTimestep();
+			if (time.doubleValue() < 0) {
+				final String errorMessage = String.format(
+					"The intervention %s has a timestep %s which is less than 0.",
+					this.getName(),
+					time.toString()
+				);
+				throw new Exception(errorMessage);
+			}
+			if (i > 0) {
+				final String appliedToOne = this.staticInterventions.get(i).getAppliedTo();
+				final Number timeOne = this.staticInterventions.get(i).getTimestep();
+				final String appliedToTwo = this.staticInterventions.get(i - 1).getAppliedTo();
+				final Number timeTwo = this.staticInterventions.get(i - 1).getTimestep();
```

Comment:
Generally speaking, it is more common to see people do 

```
for (int i = 0; i < this.staticInterventions.size() - 1; i++) {
```

Rather than looping the entire list and have a special if-clause to check consecutive elements.

Also generally speaking if you want to check for duplicates, people tend to just use Map or Set instead of this kind of iteration.

```
final Set<String> blah = new HashSet();
for (int i = 0; i < this.staticInterventions.size(); i++) {
  final String key = ....
  if (blah.contains(key)) {
    // duplicate
  } else {
    blah.add(key)
  } 
```

Reply:
Good idea.

Reply:
Thanks again for this!

Reply:
You do not even need to check for duplicate, just add into the Set, it will take care of being only one of each.

Reply:
I am checking here so that if we do find a duplicate we can propagate that error up to the user
If youre suggesting to change the datastructure of intervention to be a set instead of a list i think that will lead to confusion when the user tries to save the following:

![image](https://github.com/user-attachments/assets/50272fd3-1f44-4411-8c91-a13322ae1800)